apollo-router-core/src/plugin_utils/mock/subgraph.rs

```diff
@@ -35,6 +35,11 @@ impl Service<SubgraphRequest> for MockSubgraph {
         let response = if let Some(response) = self.mocks.get(req.http_request.body()) {
             builder.data(response.data.clone()).build().into()
         } else {
+            tracing::error!(
+                "could not find mock for query: {}",
+                serde_json::to_string(req.http_request.body())
+                    .expect("could not serialise request")
+            );
```

Comment:
can we make it a failfast debug? a panic here would be handy in tests / benches

Reply:
Will do, although I wonder if this would be better handled by mockall once the api refactor happens. Mockall give the user a decent error message if an expectation is not met. Let's discuss in a separate ticket though.

Reply:
I actually had to revert all of this. It turns out that some of our tests rely on this behaviour of returning a dummy response if not mocked :(.
We should revisit pending the refactor around apis.